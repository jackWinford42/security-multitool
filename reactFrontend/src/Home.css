#rangeBar {
    position: fixed;
    left: 20px;
    top: 100px;
}

ul {
    list-style: none;
}