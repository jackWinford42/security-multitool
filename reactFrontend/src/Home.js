import React, { useEffect, useState } from "react";
import { v4 as uuidv4 } from "uuid";
import RamtApi from "./Api";

export default function Home() {
  const [isLoading, setIsLoading] = useState(false);
  const [empty, setEmpty] = useState(true);
  const [displayHistory, setDisplayHistory] = useState([]);

  useEffect(() => {
    async function getHistory() {
      try {
        const siteHistory = await RamtApi.getSiteHistory();
        setDisplayHistory(siteHistory.history.rows);
        setEmpty(false);
      } catch (err) {
        console.error("SiteHistory get: issue loading history", err);
      }
    }
    setIsLoading(true);
    if (empty) getHistory();
    setIsLoading(false);
  }, [empty])
  console.log(displayHistory)
  const HistoryItems = displayHistory.map(row => {
    return (
    <li key={uuidv4()}>
      <h5>{row.item}</h5>
      <p>type: {row.type}</p>
      <p>safety score out of 100: {row.score}</p>
    </li>
    );
  })

  if (isLoading) return <p>Loading &hellip;</p>;
  return (
    <div className="Home">
      <ul>
        {HistoryItems}
      </ul>
    </div>
  );
}