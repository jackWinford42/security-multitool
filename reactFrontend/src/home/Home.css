#rangeBar {
    position: fixed;
    left: 20px;
    top: 100px;
}

ul {
    list-style: none;
}

.history {
    width: 700px;
    margin: 0 auto;
}

.historyGraph {
    width: 66vw;
    margin: 0 auto;
}

.email {
    padding: 5px;
    border: solid black 1px;
    background-color: violet;
    margin-bottom: 10px;
}

.url {
    padding: 5px;
    border: solid black 1px;
    background-color: cyan;
}

.BarGraph {
    padding-top: 50px;
}