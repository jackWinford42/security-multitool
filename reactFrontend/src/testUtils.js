import React from "react";
import { Provider, useDispatch } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducers/rootReducer";
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from "redux-thunk";

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(thunk)));

const demoUser = {
  email: "jackwinford@gmail.com",
  username: "captainjack",
  password: "qwerty",
  photo_url: "https://bit.ly/3C2VUgQ",
};

// const dispatch = useDispatch();
// await dispatch({type: "BEGIN_AUTH_SESSION", user: demoUser});

const UserProvider =(
    ({children}) =>
    (<Provider store={store}>
      {children}
    </Provider>)
);

export { UserProvider };