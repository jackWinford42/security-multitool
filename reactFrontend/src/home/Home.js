import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import LocalFireDepartmentIcon from '@mui/icons-material/LocalFireDepartment';
import HourglassBottomOutlinedIcon from '@mui/icons-material/HourglassBottomOutlined';
import HistoryOutlinedIcon from '@mui/icons-material/HistoryOutlined';
import ToggleButton from '@mui/material/ToggleButton';
import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';
import Tooltip from '@mui/material/Tooltip';
import "./Home.css";
import AllTime from "./AllTime";
import Hot from "./Hot";
import History from "./History";

export default React.memo(function Home() {
  const user = useSelector(st => st.currUser);
  const prevLoc = useSelector(st => st.prevLocation)
  const dispatch = useDispatch();
  const [open, setOpen] = useState(false)
  const [view, setView] = useState('history');

  //only display the welcome dialog box if the user just signed in
  if (prevLoc === "/login" || prevLoc === "/sign-up") {
    setOpen(true);
    dispatch({type: "LOCATION_CHANGE", location: "/home"})
  }

  const handleChange = (event, nextView) => {
    setView(nextView);
    console.log(nextView)
  };

  const handleClose = () => {
    setOpen(false)
  };

  const choseRange = () => {
    if (view === "history" ) {
      return (<History/>)
    } else if (view === "hot") {
      return (<Hot/>)
    } else {
      return (<AllTime/>)
    }
  }

  return (
    <div className="Home">
      <ToggleButtonGroup
        id="rangeBar"
        orientation="vertical"
        value={view}
        exclusive
        onChange={handleChange}
      >
        <Tooltip title="Recent History" placement="right">
          <ToggleButton value="history" aria-label="history">
            <HistoryOutlinedIcon/>
          </ToggleButton>        
        </Tooltip>
        <Tooltip title="Hot" placement="right">
          <ToggleButton value="hot" aria-label="hot">
            <LocalFireDepartmentIcon/>
          </ToggleButton>
        </Tooltip>
        <Tooltip title="All Time" placement="right">
          <ToggleButton value="allTime" aria-label="allTime">
            <HourglassBottomOutlinedIcon/>
          </ToggleButton>
        </Tooltip>
      </ToggleButtonGroup>
      {choseRange()}
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="dialog-title"
        aria-describedby="dialog-description"
      >
        <DialogTitle id="dialog-title">
          Welcome back {user.username}
        </DialogTitle>
        <DialogContent>
          <DialogContentText id="dialog-description">
            You are now authorized to use the Reputation Analyses Multi Tool.<br></br>
            Please, enjoy your stay.
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} autoFocus>
            Let's go
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
})
