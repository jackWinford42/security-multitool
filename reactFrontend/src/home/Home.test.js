import React from "react";
import { render } from "@testing-library/react";
import Home from "./Home";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils"

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Home/>
      </UserProvider>
    </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Home/>
      </UserProvider>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});