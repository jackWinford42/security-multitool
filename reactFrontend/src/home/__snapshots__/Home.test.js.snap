// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <div
      class="MuiToggleButtonGroup-root MuiToggleButtonGroup-vertical css-mjlvw-MuiToggleButtonGroup-root"
      id="rangeBar"
      role="group"
    >
      <button
        aria-label="history"
        aria-pressed="true"
        class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="history"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="HistoryOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.25 2.52.77-1.28-3.52-2.09V8z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="hot"
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="hot"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LocalFireDepartmentIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19.48 12.35c-1.57-4.08-7.16-4.3-5.81-10.23.1-.44-.37-.78-.75-.55C9.29 3.71 6.68 8 8.87 13.62c.18.46-.36.89-.75.59-1.81-1.37-2-3.34-1.84-4.75.06-.52-.62-.77-.91-.34C4.69 10.16 4 11.84 4 14.37c.38 5.6 5.11 7.32 6.81 7.54 2.43.31 5.06-.14 6.95-1.87 2.08-1.93 2.84-5.01 1.72-7.69zm-9.28 5.03c1.44-.35 2.18-1.39 2.38-2.31.33-1.43-.96-2.83-.09-5.09.33 1.87 3.27 3.04 3.27 5.08.08 2.53-2.66 4.7-5.56 2.32z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="allTime"
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="allTime"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="HourglassBottomOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m18 22-.01-6L14 12l3.99-4.01L18 2H6v6l4 4-4 3.99V22h12zM8 7.5V4h8v3.5l-4 4-4-4z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <ul
      class="history"
    />
  </div>
</DocumentFragment>
`;
